system/mp/mpcomm.cpp

```diff
@@ -988,30 +1051,44 @@ protected: friend class CMPPacketReader;
                             e->Release();
                         }
                     }
-#ifdef _FULLTRACE
-                    PROGLOG("MP: rd = %d", rd);
-#endif
-                    /* NB: legacy clients that don't handle the exception deserialization here
-                     * will see reply as success, so no clean error,
-                     * but will fail shortly afterwards since server connection is closed
-                     */
-                    if (rd > sizeof(rd)) // legacy clients will only ever send a reply of 0 or 4, if greater, then new client is replying with an exception
+                } // while loopcnt
+
+                /* NB: legacy clients that don't handle the exception deserialization here
+                 * will see reply as success, so no clean error,
+                 * but will fail shortly afterwards since server connection is closed
+                 */
+                if (rd == sizeof(rd))
+                {
+                    int replyVal;
+                    memcpy(&replyVal, (int *)replyBuf, sizeof(int));
```

Comment:
I want to read into an array to handle any possible amount since we are reading in intervals of an overall timeout.
I agree one is sizeof(rd) and another is sizeof(int), I will change to match.